src/backends/interflop-ieee/interflop_ieee.c

```diff
@@ -139,215 +150,159 @@ void debug_print(void *context, char *fmt_flt, char *fmt, ...) {
     }                                                                          \
   }
 
-inline void debug_print_float(void *context, operation_type typeop,
-                              const char *op, float a, float b, float c) {
+inline void debug_print_float(void *context, const operation_type typeop,
+                              const char *op, const float a, const float b,
+                              const float c) {
   DEBUG_PRINT(context, typeop, op, a, b, c);
 }
 
-inline void debug_print_double(void *context, operation_type typeop,
-                               const char *op, double a, double b, double c) {
+inline void debug_print_double(void *context, const operation_type typeop,
+                               const char *op, const double a, const double b,
+                               const double c) {
   DEBUG_PRINT(context, typeop, op, a, b, c);
 }
 
-static void _interflop_add_float(float a, float b, float *c, void *context) {
+/* Set C to the result of the comparison P between A and B */
+/* Set STR to name of the comparison  */
+#define SELECT_FLOAT_CMP(A, B, C, P, STR)                                      \
+  switch (P) {                                                                 \
+  case FCMP_FALSE:                                                             \
+    *C = false;                                                                \
+    STR = "FCMP_FALSE";                                                        \
+    break;                                                                     \
+  case FCMP_OEQ:                                                               \
+    *C = ((!isnan(A)) && (!isnan(B)) && (A == B));                             \
```

Comment:
This can be simplified. We do not need the isnan tests which are anyways done in IEEE-754 except for unordered and ordered